<p>
 <img src="ayiti_analytics.png" alt="Ayiti Analytics"width=400 />
</p>

# Capstone Project - The Battle of the Neighborhoods (Week 2)
### Applied Data Science Capstone by IBM/Coursera

## Table of contents
* [Introduction: Business Problem](#introduction)
* [Data](#data)
* [Methodology](#methodology)
* [Analysis](#analysis)
* [Results and Discussion](#results)
* [Conclusion](#conclusion)

## Introduction: Business Problem <a name="introduction"></a>

In this project we will try How to find a Dance School next to an Elementary school in **Chicago** with a **high score of Instruction** and where **the teachers score** and **the safety score** is also high. These three parameters are very important because it is important for a parent to know that his or her kid is safe at school, that the instruction level is great and the teachers score is good enough.

## Data <a name="data"></a>

Based on definition of our problem, factors that will influence our decission are:
* Number of School Dance in Chicago
* Number 

In [1]:
import pandas as pd
import numpy as np
from geopy.geocoders import Nominatim

import json

import requests
from pandas.io.json import json_normalize

import matplotlib.cm as cm
import matplotlib.colors as colors
import matplotlib.cm as cm
from sklearn.cluster import KMeans

import folium

In [2]:
df_capstone = pd.read_csv("Chicago_Public_Schools.csv")

In [3]:
neigh = pd.read_csv('Neighborhoods_2012b.csv')
neig_list = list(set(neigh.PRI_NEIGH))

## Methodology <a name="methodology"></a>

After having done our data processing import our libraries look at the types of each column on jupyter notebook calculate their average describes our numeric columns. our dataset was at about 78 columns after our analysis it was reduced to those columns which are necessary for us:
* NAME_OF_SCHOOL
* Elementary,Middle, or High School
* SAFETY_SCORE
* Instruction Score
* Teachers Score
* Latitude
* Longitude
* COMMUNITY_AREA_NAME
* Location
* Street Address

**We requested data on dance schools located in chicago thanks to foursquare in order to achieve our goal of finding a primary school with a high level of education, high scoring teachers and a high safety score next to a school dance.**
**We have created maps to see the city of Chicago and the addresses of the Schools.**


In [4]:
df_capstone.head()

School ID                                     NAME_OF_SCHOOL  \
0     610038                  Abraham Lincoln Elementary School   
1     610281  Adam Clayton Powell Paideia Community Academy ...   
2     610185                Adlai E Stevenson Elementary School   
3     609993                    Agustin Lara Elementary Academy   
4     610513                      Air Force Academy High School   

  Elementary, Middle, or High School          Street Address     City State  \
0                                 ES        615 W Kemper Pl   Chicago    IL   
1                                 ES  7511 S South Shore Dr   Chicago    IL   
2                                 ES     8010 S Kostner Ave   Chicago    IL   
3                                 ES     4619 S Wolcott Ave   Chicago    IL   
4                                 HS        3630 S Wells St   Chicago    IL   

   ZIP Code    Phone Number  \
0     60614  (773) 534-5720   
1     60649  (773) 535-6650   
2     60652  (773) 535-2280   
3     60609  (773) 535-4389   
4     60609  (773) 535-1590   

                                               Link   \
0  http://schoolreports.cps.edu/SchoolProgressRep...   
1  http://schoolreports.cps.edu/SchoolProgressRep...   
2  http://schoolreports.cps.edu/SchoolProgressRep...   
3  http://schoolreports.cps.edu/SchoolProgressRep...   
4  http://schoolreports.cps.edu/SchoolProgressRep...   

                      Network Manager  ... Freshman on Track Rate %  \
0        Fullerton Elementary Network  ...                      NDA   
1           Skyway Elementary Network  ...                      NDA   
2           Midway Elementary Network  ...                      NDA   
3         Pershing Elementary Network  ...                      NDA   
4  Southwest Side High School Network  ...                     91.8   

  X_COORDINATE Y_COORDINATE   Latitude  Longitude COMMUNITY_AREA_NUMBER  \
0  1171699.458  1915829.428  41.924497 -87.644522                     7   
1  1196129.985  1856209.466  41.760324 -87.556736                    43   
2  1148427.165  1851012.215  41.747111 -87.731702                    70   
3  1164504.290  1873959.199  41.809757 -87.672145                    61   
4  1175177.622  1880745.126  41.828146 -87.632794                    34   

  COMMUNITY_AREA_NAME  Ward Police District                     Location  
0        LINCOLN PARK    43              18  (41.92449696, -87.64452163)  
1         SOUTH SHORE     7               4  (41.76032435, -87.55673627)  
2             ASHBURN    13               8  (41.74711093, -87.73170248)  
3            NEW CITY    20               9    (41.8097569, -87.6721446)  
4       ARMOUR SQUARE    11               9  (41.82814609, -87.63279369)  

[5 rows x 78 columns]

In [5]:
df_capstone.columns

Index(['School ID', 'NAME_OF_SCHOOL', 'Elementary, Middle, or High School',
       'Street Address', 'City', 'State', 'ZIP Code', 'Phone Number', 'Link ',
       'Network Manager', 'Collaborative Name',
       'Adequate Yearly Progress Made? ', 'Track Schedule',
       'CPS Performance Policy Status', 'CPS Performance Policy Level',
       'HEALTHY_SCHOOL_CERTIFIED', 'Safety Icon ', 'SAFETY_SCORE',
       'Family Involvement Icon', 'Family Involvement Score',
       'Environment Icon ', 'Environment Score', 'Instruction Icon ',
       'Instruction Score', 'Leaders Icon ', 'Leaders Score ',
       'Teachers Icon ', 'Teachers Score', 'Parent Engagement Icon ',
       'Parent Engagement Score', 'Parent Environment Icon',
       'Parent Environment Score', 'AVERAGE_STUDENT_ATTENDANCE',
       'Rate of Misconducts (per 100 students) ', 'Average Teacher Attendance',
       'Individualized Education Program Compliance Rate ', 'Pk-2 Literacy %',
       'Pk-2 Math %', 'Gr3-5 Grade Level Math %'

# Analysis <a name="analysis"></a>

In [6]:
df_capstone.mean()

School ID                                  6.096818e+05
ZIP Code                                   6.062973e+04
SAFETY_SCORE                               4.950487e+01
Environment Score                          4.776608e+01
Instruction Score                          4.828850e+01
Rate of Misconducts (per 100 students)     2.108587e+01
ISAT Exceeding Math %                      2.035630e+01
ISAT Exceeding Reading %                   1.566702e+01
ISAT Value Add Math                        6.858974e-02
ISAT Value Add Read                        2.264957e-02
COLLEGE_ENROLLMENT                         6.260530e+02
General Services Route                     3.908481e+01
X_COORDINATE                               1.163016e+06
Y_COORDINATE                               1.885663e+06
Latitude                                   4.184189e+01
Longitude                                 -8.767732e+01
COMMUNITY_AREA_NUMBER                      3.770671e+01
Ward                                       2.206

In [7]:
df_capstone.describe()

School ID      ZIP Code  SAFETY_SCORE  Environment Score  \
count     566.000000    566.000000    513.000000         513.000000   
mean   609681.791519  60629.727915     49.504873          47.766082   
std      8831.060946     20.248024     20.110837          16.215584   
min    400018.000000  60605.000000      1.000000           1.000000   
25%    609873.250000  60618.000000     35.000000          37.000000   
50%    610052.500000  60625.000000     48.000000          47.000000   
75%    610212.750000  60639.000000     61.000000          58.000000   
max    610544.000000  60827.000000     99.000000          99.000000   

       Instruction Score  Rate of Misconducts (per 100 students)   \
count         513.000000                               566.000000   
mean           48.288499                                21.085866   
std            17.417176                                27.543835   
min             1.000000                                 0.000000   
25%            37.000000                                 4.525000   
50%            47.000000                                12.250000   
75%            59.000000                                26.975000   
max            99.000000                               251.600000   

       ISAT Exceeding Math %  ISAT Exceeding Reading %   ISAT Value Add Math  \
count             476.000000                 476.000000           468.000000   
mean               20.356303                  15.667017             0.068590   
std                16.850870                  15.208361             1.041174   
min                 0.000000                   0.000000            -3.500000   
25%                 9.175000                   6.100000            -0.600000   
50%                16.000000                  10.950000             0.000000   
75%                24.900000                  19.525000             0.700000   
max               100.000000                 100.000000             3.600000   

       ISAT Value Add Read  COLLEGE_ENROLLMENT  General Services Route   \
count           468.000000          566.000000               566.000000   
mean              0.022650          626.053004                39.084806   
std               1.042543          448.495105                 6.165552   
min              -5.000000           21.000000                29.000000   
25%              -0.600000          342.250000                34.000000   
50%               0.000000          504.000000                39.000000   
75%               0.625000          790.500000                45.000000   
max               4.900000         4368.000000                49.000000   

       X_COORDINATE  Y_COORDINATE    Latitude   Longitude  \
count  5.660000e+02  5.660000e+02  566.000000  566.000000   
mean   1.163016e+06  1.885663e+06   41.841890  -87.677323   
std    1.605930e+04  3.285688e+04    0.090361    0.058390   
min    1.118114e+06  1.817242e+06   41.653674  -87.841052   
25%    1.152166e+06  1.860248e+06   41.771866  -87.717152   
50%    1.163553e+06  1.886889e+06   41.845521  -87.675878   
75%    1.174256e+06  1.911652e+06   41.913584  -87.636059   
max    1.202811e+06  1.950960e+06   42.021064  -87.533007   

       COMMUNITY_AREA_NUMBER        Ward  Police District  
count             566.000000  566.000000       566.000000  
mean               37.706714   22.060071        11.710247  
std                21.584135   13.645574         6.688582  
min                 1.000000    1.000000         1.000000  
25%                23.000000   10.000000         7.000000  
50%                31.500000   21.000000        10.000000  
75%                58.000000   32.750000        17.000000  
max                77.000000   50.000000        25.000000

In [8]:
df_capstone2 = df_capstone [['NAME_OF_SCHOOL','Street Address','Elementary, Middle, or High School','SAFETY_SCORE','Instruction Score','Teachers Score','Latitude','Longitude','COMMUNITY_AREA_NAME','Location']]

In [9]:
df_capstone2_ = df_capstone2[df_capstone2["Elementary, Middle, or High School"] == 'ES']

In [10]:
df_capstone2_ = df_capstone2_[df_capstone2_["SAFETY_SCORE"] >= 70.0]

In [11]:
df_capstone2_= df_capstone2_[df_capstone2_["Instruction Score"] >= 70.0]

In [12]:
df_capstone2_= df_capstone2_[df_capstone2_["Teachers Score"] >= '70.0']

In [13]:
df_capstone2_.describe()

SAFETY_SCORE  Instruction Score   Latitude  Longitude
count      9.000000           9.000000   9.000000   9.000000
mean      95.222222          88.666667  41.892787 -87.703842
std        7.758508          11.800424   0.098582   0.043923
min       78.000000          71.000000  41.697198 -87.812091
25%       99.000000          80.000000  41.858370 -87.704655
50%       99.000000          95.000000  41.896740 -87.690298
75%       99.000000          99.000000  41.958017 -87.683179
max       99.000000          99.000000  42.009307 -87.664958

In [14]:
df_capstone2_.shape

(9, 10)

In [15]:
df_capstone2.describe()

SAFETY_SCORE  Instruction Score    Latitude   Longitude
count    513.000000         513.000000  566.000000  566.000000
mean      49.504873          48.288499   41.841890  -87.677323
std       20.110837          17.417176    0.090361    0.058390
min        1.000000           1.000000   41.653674  -87.841052
25%       35.000000          37.000000   41.771866  -87.717152
50%       48.000000          47.000000   41.845521  -87.675878
75%       61.000000          59.000000   41.913584  -87.636059
max       99.000000          99.000000   42.021064  -87.533007

In [16]:
# the types of our dataset
df_capstone2.dtypes

NAME_OF_SCHOOL                         object
Street Address                         object
Elementary, Middle, or High School     object
SAFETY_SCORE                          float64
Instruction Score                     float64
Teachers Score                         object
Latitude                              float64
Longitude                             float64
COMMUNITY_AREA_NAME                    object
Location                               object
dtype: object

In [17]:
# we can see the mean
df_capstone2.mean()

SAFETY_SCORE         49.504873
Instruction Score    48.288499
Latitude             41.841890
Longitude           -87.677323
dtype: float64

In [18]:
df_capstone2.COMMUNITY_AREA_NAME = df_capstone2.COMMUNITY_AREA_NAME.str.lower()
df_capstone2

C:\Users\bootcamp\Anaconda3\lib\site-packages\pandas\core\generic.py:5208: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value


NAME_OF_SCHOOL  \
0                    Abraham Lincoln Elementary School   
1    Adam Clayton Powell Paideia Community Academy ...   
2                  Adlai E Stevenson Elementary School   
3                      Agustin Lara Elementary Academy   
4                        Air Force Academy High School   
..                                                 ...   
561                William T Sherman Elementary School   
562                 William W Carter Elementary School   
563                Wolfgang A Mozart Elementary School   
564               Woodlawn Community Elementary School   
565                 World Language Academy High School   

             Street Address Elementary, Middle, or High School  SAFETY_SCORE  \
0          615 W Kemper Pl                                  ES          99.0   
1    7511 S South Shore Dr                                  ES          54.0   
2       8010 S Kostner Ave                                  ES          61.0   
3       4619 S Wolcott Ave                                  ES          56.0   
4          3630 S Wells St                                  HS          49.0   
..                      ...                                ...           ...   
561         1000 W 52nd St                                  ES          32.0   
562    5740 S Michigan Ave                                  ES          13.0   
563      2200 N Hamlin Ave                                  ES          41.0   
564     6657 S Kimbark Ave                                  ES          70.0   
565     3120 S Kostner Ave                                  HS          51.0   

     Instruction Score Teachers Score   Latitude  Longitude  \
0                 66.0             70  41.924497 -87.644522   
1                 84.0             76  41.760324 -87.556736   
2                 36.0            NDA  41.747111 -87.731702   
3                 37.0             48  41.809757 -87.672145   
4                 55.0             54  41.828146 -87.632794   
..                 ...            ...        ...        ...   
561               55.0            NDA  41.799788 -87.650255   
562               35.0             62  41.789841 -87.622490   
563               32.0            NDA  41.920927 -87.721925   
564               66.0            NDA  41.773400 -87.594356   
565               47.0             49  41.836020 -87.734195   

    COMMUNITY_AREA_NAME                     Location  
0          lincoln park  (41.92449696, -87.64452163)  
1           south shore  (41.76032435, -87.55673627)  
2               ashburn  (41.74711093, -87.73170248)  
3              new city    (41.8097569, -87.6721446)  
4         armour square  (41.82814609, -87.63279369)  
..                  ...                          ...  
561            new city  (41.79978772, -87.65025483)  
562     washington park  (41.78984129, -87.62248974)  
563        logan square  (41.92092734, -87.72192541)  
564            woodlawn  (41.77339962, -87.59435584)  
565      south lawndale  (41.83601953, -87.73419465)  

[566 rows x 10 columns]

## Clustering

In [19]:
longitudes=[]
latitudes= []
neighs= []
for index in range(0,len(neig_list)):
    print(index)
    try:
        print(neig_list[index])
        query = neig_list[index]+ " Chicago,IL"
        geolocator = Nominatim(user_agent="ny_explorer")
        location = geolocator.geocode(query)
        latitudes.append(location.latitude)
        longitudes.append(location.longitude)
        neighs.append(neig_list[index])
    except:

        latitudes.append(np.nan)
        longitudes.append(np.nan)
        neighs.append(neig_list[index])
        
df_3 =pd.DataFrame(dict(Neighborhoods=neighs,latitude=latitudes,longitude=longitudes))

0
Lower West Side
1
Avalon Park
2
Mckinley Park
3
Chicago Lawn
4
Fuller Park
5
Sauganash,Forest Glen
6
Dunning
7
Galewood
8
Printers Row
9
Jackson Park
10
Garfield Ridge
11
Oakland
12
Roseland
13
Hegewisch
14
Old Town
15
Grand Crossing
16
Morgan Park
17
Lincoln Park
18
Douglas
19
North Center
20
North Lawndale
21
Greektown
22
Edgewater
23
Chatham
24
Magnificent Mile
25
Humboldt Park
26
Wrigleyville
27
Grand Boulevard
28
Edison Park
29
Lincoln Square
30
Ukrainian Village
31
East Village
32
Millenium Park
33
Museum Campus
34
United Center
35
Near South Side
36
Belmont Cragin
37
Hyde Park
38
Uptown
39
New City
40
West Pullman
41
Bucktown
42
Armour Square
43
West Lawn
44
Woodlawn
45
Norwood Park
46
Andersonville
47
Garfield Park
48
Pullman
49
Mount Greenwood
50
Clearing
51
Jefferson Park
52
Englewood
53
South Deering
54
East Side
55
Washington Heights
56
Bridgeport
57
Streeterville
58
Portage Park
59
River North
60
O'Hare
61
Montclare
62
Grant Park
63
Gage Park
64
South Shore
65
Sheffield 

In [20]:
latitudes
longitudes

[-87.6656086,
 -87.5886584,
 -87.6736638,
 -87.6964408,
 -87.6325508,
 -87.73523436690189,
 -87.7964493,
 -87.7928372,
 -87.6288996,
 -87.63575316733329,
 -87.74548862491682,
 -87.6082424,
 -87.625881,
 -87.5469885,
 -87.6553579,
 -87.6003258,
 -87.6667156,
 -87.647832,
 -87.6179536,
 -87.6791596,
 -87.7183883,
 -87.6469318,
 -87.6639516,
 -87.6125481,
 -87.624228,
 -87.7041737,
 -87.6564773,
 -87.6172724,
 -87.81401633833357,
 -87.6896163305115,
 -87.6847499,
 -87.6244212,
 nan,
 120.9753251,
 -87.67418510441388,
 -87.6247738,
 -87.7686699,
 -87.5939244,
 -87.6555458,
 -87.6564403,
 -87.637823,
 -87.6701232,
 -87.633107,
 -87.722275,
 -87.599493,
 -87.80058173001102,
 -87.66927341557962,
 -87.71591710895399,
 -87.6066039,
 -87.7086616,
 -87.7733881,
 -87.7631179,
 -87.6458836,
 -87.57170331249044,
 -87.5333872,
 -87.65616,
 -87.651028,
 -87.6219967,
 -87.7650594,
 -87.8411818,
 -87.90676788547228,
 -87.8008931,
 -87.61928550074734,
 -87.6961636,
 -87.5752828,
 nan,
 -87.6244212,
 -87.

In [21]:
df_3

Neighborhoods   latitude  longitude
0   Lower West Side  41.854200 -87.665609
1       Avalon Park  41.745035 -87.588658
2     Mckinley Park  41.831700 -87.673664
3      Chicago Lawn  41.775033 -87.696441
4       Fuller Park  41.818089 -87.632551
..              ...        ...        ...
93   Little Village  41.875562 -87.624421
94         Avondale  41.938921 -87.711168
95  Calumet Heights  41.730035 -87.579213
96          Beverly  41.718153 -87.671767
97      Wicker Park  41.907802 -87.678155

[98 rows x 3 columns]

In [22]:
#delete null values
df_3.dropna(inplace=True)

In [23]:
df_3.Neighborhoods =df_3.Neighborhoods.str.lower()
df_3.head()

Neighborhoods   latitude  longitude
0  lower west side  41.854200 -87.665609
1      avalon park  41.745035 -87.588658
2    mckinley park  41.831700 -87.673664
3     chicago lawn  41.775033 -87.696441
4      fuller park  41.818089 -87.632551

## Foursquare

In [24]:
client_id = '0KTKKCYIO0BFEWJMDQ4ZTOADCGOED1KYXWIVS1JO3DPQ0LG1' # our Foursquare ID
client_secret = '5URQZPCQJT4U1ZOU1I1NX2AUYIWBFNVHC520MXQE3BFZRJYW' # our Foursquare Secret
version = '20180604'
radius = 3000
query= "Chicago,IL"                                                  



limit=500
categoryId ="4bf58dd8d48988d1f2931735"
url_root = " https://api.foursquare.com/v2/venues/search?"

In [25]:
# convert an address into latitude and longitude values
geolocator = Nominatim(user_agent="ny_explorer")
location = geolocator.geocode(query)
latitude = location.latitude
longitude = location.longitude
(latitude,longitude)

(41.8755616, -87.6244212)

In [26]:
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}&categoryId={}'.format(client_id, client_secret, latitude, longitude, version, query, radius, limit,categoryId)
url

'https://api.foursquare.com/v2/venues/search?client_id=0KTKKCYIO0BFEWJMDQ4ZTOADCGOED1KYXWIVS1JO3DPQ0LG1&client_secret=5URQZPCQJT4U1ZOU1I1NX2AUYIWBFNVHC520MXQE3BFZRJYW&ll=41.8755616,-87.6244212&v=20180604&query=Chicago,IL&radius=3000&limit=500&categoryId=4bf58dd8d48988d1f2931735'

In [27]:
#show the dataset found in foursquare
results = requests.get(url).json()
results['response']['venues']

[{'id': '5a7b7784b9ac383f9f6621d6',
  'name': 'American Rhythm Center, 410 S Michigan Ave, 3Rd Floor, Chicago, Il',
  'location': {'address': '410 S Michigan Ave',
   'lat': 41.876447999999996,
   'lng': -87.62485699999999,
   'labeledLatLngs': [{'label': 'display',
     'lat': 41.876447999999996,
     'lng': -87.62485699999999}],
   'distance': 105,
   'cc': 'US',
   'city': 'Chicago',
   'state': 'IL',
   'country': 'United States',
   'formattedAddress': ['410 S Michigan Ave', 'Chicago, IL', 'United States']},
  'categories': [{'id': '4bf58dd8d48988d134941735',
    'name': 'Dance Studio',
    'pluralName': 'Dance Studios',
    'shortName': 'Dance Studio',
    'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/arts_entertainment/performingarts_dancestudio_',
     'suffix': '.png'},
    'primary': True}],
  'referralId': 'v-1595106346',
  'hasPerk': False},
 {'id': '4ae786a5f964a52019ac21e3',
  'name': 'Auditorium Theatre',
  'location': {'address': '50 E Congress Pkwy',
   'c

In [28]:
#let's see our venues
venues = results['response']['venues']

# tranform venues into a dataframe
dataframe = json_normalize(venues)
dataframe

id  \
0   5a7b7784b9ac383f9f6621d6   
1   4ae786a5f964a52019ac21e3   
2   4c45c0e0dcd61b8d961f7d56   
3   50175d13e4b002d58e736599   
4   50177c8de4b0d51f8e12ca0a   
5   4aa05f40f964a520643f20e3   
6   4dbb90fcbe4d6d4c08fc896f   
7   4adfca6df964a520777d21e3   
8   5036ca0be4b0e82a0177daf0   
9   4a97023cf964a520f32720e3   
10  4cbf2ac1575d236aa846bd8e   
11  532e0abb498e781e73bd556c   
12  57be9d8d498e384fe4eaa1c1   
13  4ae06c42f964a5202b7f21e3   
14  4d364557cea8b60c8d7917f8   
15  4b284c70f964a5202f9224e3   
16  5a282e85c9a517611103784e   
17  4d2b2852ebacb1f71ba90950   
18  4f08afc9e4b026656e7f7a7e   
19  584267bd266c11029e81cea0   
20  4e31e29bb3adb5d1a87e0b75   
21  57044dcc498e0ff657f223fc   
22  5944dae72be4250e31d3cd8b   
23  566c466a498e45858a6bd838   
24  4d6c6096b236a143c6ba46c4   
25  582a0e661632352ed5a8e89b   
26  51415973e4b0876e26b99b2b   
27  4b4b8bf4f964a520d59f26e3   
28  57a45ba7498e7269d63bc5ca   
29  53054bc4498eabc0c6238898   
30  5062fc08e4b0b8d63d96e80b   
31  59ed527116fa044e98292cbd   
32  5081e0a5e4b03db175ca369a   
33  4beb6d4f9fa3ef3ba23d80c9   
34  51d09bba498e98300a214039   
35  4bbf636985fbb71383187267   
36  5d8702286931ce0008494d20   
37  4b464ee1f964a5202a1d26e3   
38  4b929624f964a5200a0734e3   
39  4df5008eaeb7170aa2f647ae   
40  4b2189c5f964a5207b3d24e3   
41  4b10c31df964a520837523e3   
42  4c34f577a0ced13ac40f196e   
43  54654b39498e762c1d83f5de   
44  4b6cc97df964a52061542ce3   
45  5b26afcc5c6838002c5c559a   
46  4cd725fc122ba143a00837a1   
47  593a0085f0b4902194be6718   
48  514d2c66e4b0a1833734c100   
49  4bad6c2ef964a520634e3be3   

                                                 name  \
0   American Rhythm Center, 410 S Michigan Ave, 3R...   
1                                  Auditorium Theatre   
2                   Chicago Youth Symphony Orchestras   
3   Chicago Human Rhythm Project (@ The Fine Arts ...   
4   Chicago Human Rhythm Project - American Rhythm...   
5                                 The Chicago Theatre   
6                               Chicago Jazz Ensemble   
7        Symphony Center (Chicago Symphony Orchestra)   
8                  Chicago Summer Dance in Grant Park   
9                         Chicago Shakespeare Theater   
10                                       Chicago Muse   
11                 Chicago Multicultural Dance Center   
12                                    Chicago Theatre   
13                     Chicago Tribune Freedom Center   
14                 Chicago Multicultural Dance Center   
15                            Cadillac Palace Theatre   
16                                The Chicago Theatre   
17                              Chicago Opera Theater   
18                Chicago Shakespeare Rehearsal Space   
19                                The Chicago Theatre   
20                              Chicago Opera Theater   
21                            Chicago Performing Arts   
22                        PrivateBank Theatre Chicago   
23                         Chicago Children's Theatre   
24                         Civic Orchestra Of Chicago   
25                              Chicago Dance Academy   
26                                  Chicago Lindy Hop   
27                                      Chicago Dance   
28                                      Chicago Dance   
29                         chicago urban arts society   
30                                Chicago Fire TV Set   
31          Chicago Children’s Theatre At The Station   
32                         Chicago Children's Theatre   
33             Guitar Chicago (Guitar Cities Chicago)   
34                                        SummerDance   
35                     Chicago Center School Of Music   
36                        Drunk Shakespeare - Chicago   
37             ShowPlace ICON at Roosevelt Collection   
38                                     Ballet Chicago   
39                             Mp3 Experiment Chicago   
40                         League of Chicago Theatres   

In [29]:
# keep only columns that include venue name, and anything that is associated with location
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
dataframe_filtered = dataframe.loc[:, filtered_columns]

# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

# filter the category for each row
dataframe_filtered['categories'] = dataframe_filtered.apply(get_category_type, axis=1)

# clean column names by keeping only last term
dataframe_filtered.columns = [column.split('.')[-1] for column in dataframe_filtered.columns]

dataframe_filtered

name             categories  \
0   American Rhythm Center, 410 S Michigan Ave, 3R...           Dance Studio   
1                                  Auditorium Theatre                Theater   
2                   Chicago Youth Symphony Orchestras  Performing Arts Venue   
3   Chicago Human Rhythm Project (@ The Fine Arts ...           Dance Studio   
4   Chicago Human Rhythm Project - American Rhythm...           Dance Studio   
5                                 The Chicago Theatre                Theater   
6                               Chicago Jazz Ensemble  Performing Arts Venue   
7        Symphony Center (Chicago Symphony Orchestra)           Concert Hall   
8                  Chicago Summer Dance in Grant Park  Performing Arts Venue   
9                         Chicago Shakespeare Theater                Theater   
10                                       Chicago Muse                Theater   
11                 Chicago Multicultural Dance Center           Dance Studio   
12                                    Chicago Theatre                Theater   
13                     Chicago Tribune Freedom Center               Building   
14                 Chicago Multicultural Dance Center           Dance Studio   
15                            Cadillac Palace Theatre                Theater   
16                                The Chicago Theatre                Theater   
17                              Chicago Opera Theater                Theater   
18                Chicago Shakespeare Rehearsal Space                Theater   
19                                The Chicago Theatre  Performing Arts Venue   
20                              Chicago Opera Theater  Performing Arts Venue   
21                            Chicago Performing Arts  Performing Arts Venue   
22                        PrivateBank Theatre Chicago                Theater   
23                         Chicago Children's Theatre                Theater   
24                         Civic Orchestra Of Chicago            Opera House   
25                              Chicago Dance Academy           Dance Studio   
26                                  Chicago Lindy Hop           Dance Studio   
27                                      Chicago Dance           Dance Studio   
28                                      Chicago Dance           Dance Studio   
29                         chicago urban arts society  Performing Arts Venue   
30                                Chicago Fire TV Set                Theater   
31          Chicago Children’s Theatre At The Station                Theater   
32                         Chicago Children's Theatre                Theater   
33             Guitar Chicago (Guitar Cities Chicago)           Music School   
34                                        SummerDance           Dance Studio   
35                     Chicago Center School Of Music  Performing Arts Venue   
36                        Drunk Shakespeare - Chicago          Indie Theater   
37             ShowPlace ICON at Roosevelt Collection              Multiplex   
38                                     Ballet Chicago           Dance Studio   
39                             Mp3 Experiment Chicago                Theater   
40                         League of Chicago Theatres                Theater   
41                                  Civic Opera House            Opera House   
42                            Civic Ballet of Chicago  Performing Arts Venue   
43   Dinner Detective Interactive Murder Mystery Show  General Entertainment   
44                                 Arie Crown Theater                Theater   
45                                           The Yard                Theater   
46                               Shaw Chicago Theatre                Theater   
47                        Ballroom Dance Chicago Loft           Dance Studio   
48                  Stage Door of the Chicago Theater                Theater   
49                         Latin Rhythms Dance St

In [30]:
# here we filtre our columns ['name','categories','address','lat','lng']
df_capstone3 = dataframe_filtered[['name','categories','address','lat','lng','neighborhood']]
df_capstone3.head()

name             categories  \
0  American Rhythm Center, 410 S Michigan Ave, 3R...           Dance Studio   
1                                 Auditorium Theatre                Theater   
2                  Chicago Youth Symphony Orchestras  Performing Arts Venue   
3  Chicago Human Rhythm Project (@ The Fine Arts ...           Dance Studio   
4  Chicago Human Rhythm Project - American Rhythm...           Dance Studio   

                      address        lat        lng neighborhood  
0          410 S Michigan Ave  41.876448 -87.624857          NaN  
1          50 E Congress Pkwy  41.876058 -87.625303          NaN  
2  410 S Michigan Ave Ste 833  41.876448 -87.624857          NaN  
3       410 S Michigan Avenue  41.876448 -87.624857          NaN  
4      410 S. Michigan Avenue  41.876448 -87.624857          NaN

# Here the maps of all  Elementary School at Chicago

In [31]:
address = 'Chicago,IL'

geolocator = Nominatim(user_agent="ny_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinates of Chicago,IL are {}, {}.'.format(latitude, longitude))

The geograpical coordinates of Chicago,IL are 41.8755616, -87.6244212.


In [32]:
 #create map of CHicago using latitude and longitude values
chicago_map = folium.Map(location=[latitude, longitude], zoom_start=10)

# add markers to map
for NAME_OF_SCHOOL, lat, lng,COMMUNITY_AREA_NAME in zip(df_capstone2_['NAME_OF_SCHOOL'], df_capstone2_['Latitude'], df_capstone2_['Longitude'],df_capstone2_['COMMUNITY_AREA_NAME']):
    label = '{},{}'.format(NAME_OF_SCHOOL,COMMUNITY_AREA_NAME)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(chicago_map)  
    
chicago_map

In [33]:
df_capstone3 = dataframe_filtered[['name','categories','address','lat','lng']]
df_capstone3

name             categories  \
0   American Rhythm Center, 410 S Michigan Ave, 3R...           Dance Studio   
1                                  Auditorium Theatre                Theater   
2                   Chicago Youth Symphony Orchestras  Performing Arts Venue   
3   Chicago Human Rhythm Project (@ The Fine Arts ...           Dance Studio   
4   Chicago Human Rhythm Project - American Rhythm...           Dance Studio   
5                                 The Chicago Theatre                Theater   
6                               Chicago Jazz Ensemble  Performing Arts Venue   
7        Symphony Center (Chicago Symphony Orchestra)           Concert Hall   
8                  Chicago Summer Dance in Grant Park  Performing Arts Venue   
9                         Chicago Shakespeare Theater                Theater   
10                                       Chicago Muse                Theater   
11                 Chicago Multicultural Dance Center           Dance Studio   
12                                    Chicago Theatre                Theater   
13                     Chicago Tribune Freedom Center               Building   
14                 Chicago Multicultural Dance Center           Dance Studio   
15                            Cadillac Palace Theatre                Theater   
16                                The Chicago Theatre                Theater   
17                              Chicago Opera Theater                Theater   
18                Chicago Shakespeare Rehearsal Space                Theater   
19                                The Chicago Theatre  Performing Arts Venue   
20                              Chicago Opera Theater  Performing Arts Venue   
21                            Chicago Performing Arts  Performing Arts Venue   
22                        PrivateBank Theatre Chicago                Theater   
23                         Chicago Children's Theatre                Theater   
24                         Civic Orchestra Of Chicago            Opera House   
25                              Chicago Dance Academy           Dance Studio   
26                                  Chicago Lindy Hop           Dance Studio   
27                                      Chicago Dance           Dance Studio   
28                                      Chicago Dance           Dance Studio   
29                         chicago urban arts society  Performing Arts Venue   
30                                Chicago Fire TV Set                Theater   
31          Chicago Children’s Theatre At The Station                Theater   
32                         Chicago Children's Theatre                Theater   
33             Guitar Chicago (Guitar Cities Chicago)           Music School   
34                                        SummerDance           Dance Studio   
35                     Chicago Center School Of Music  Performing Arts Venue   
36                        Drunk Shakespeare - Chicago          Indie Theater   
37             ShowPlace ICON at Roosevelt Collection              Multiplex   
38                                     Ballet Chicago           Dance Studio   
39                             Mp3 Experiment Chicago                Theater   
40                         League of Chicago Theatres                Theater   
41                                  Civic Opera House            Opera House   
42                            Civic Ballet of Chicago  Performing Arts Venue   
43   Dinner Detective Interactive Murder Mystery Show  General Entertainment   
44                                 Arie Crown Theater                Theater   
45                                           The Yard                Theater   
46                               Shaw Chicago Theatre                Theater   
47                        Ballroom Dance Chicago Loft           Dance Studio   
48                  Stage Door of the Chicago Theater                Theater   
49                         Latin Rhythms Dance St

In [34]:
df_capstone4 = df_capstone3[df_capstone3["categories"] == 'Dance Studio']

In [41]:
df_capstone4

name    categories  \
0   American Rhythm Center, 410 S Michigan Ave, 3R...  Dance Studio   
3   Chicago Human Rhythm Project (@ The Fine Arts ...  Dance Studio   
4   Chicago Human Rhythm Project - American Rhythm...  Dance Studio   
11                 Chicago Multicultural Dance Center  Dance Studio   
14                 Chicago Multicultural Dance Center  Dance Studio   
25                              Chicago Dance Academy  Dance Studio   
26                                  Chicago Lindy Hop  Dance Studio   
27                                      Chicago Dance  Dance Studio   
28                                      Chicago Dance  Dance Studio   
34                                        SummerDance  Dance Studio   
38                                     Ballet Chicago  Dance Studio   
47                        Ballroom Dance Chicago Loft  Dance Studio   

                         address        lat        lng  
0             410 S Michigan Ave  41.876448 -87.624857  
3          410 S Michigan Avenue  41.876448 -87.624857  
4         410 S. Michigan Avenue  41.876448 -87.624857  
11             806 S Plymouth Ct  41.871843 -87.629103  
14                  47 W Polk St  41.871843 -87.629103  
25             1920 S Wabash Ave  41.855942 -87.626069  
26                415 W Huron St  41.894485 -87.639174  
27  731 N Sangamon St, 3rd Floor  41.894675 -87.638915  
28                415 W Huron St  41.894499 -87.639189  
34            601 S Michigan Ave  41.873804 -87.623335  
38                 17 N State St  41.882693 -87.627365  
47     845 W Fulton St Suite 213  41.886607 -87.649164

In [36]:
chicago_map2 = folium.Map(location=[latitude, longitude], zoom_start=13)

# add markers to map
for name, lat, lng in zip(df_capstone4['name'], df_capstone4['lat'], df_capstone4['lng']):
    label = '{}'.format(name)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=4,
        popup=label,
        color='red',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(chicago_map2)  
    
chicago_map2

In [37]:
df_capstone4.shape

(12, 5)

In [38]:
df_capstone2_.rename(columns={"Street Address": "address","Elementary, Middle, or High School":"Grade","COMMUNITY_AREA_NAME":"borough"}, errors="raise")

NAME_OF_SCHOOL                 address  \
27       Annie Keller Elementary Gifted Magnet School        3020 W 108th St    
141                  Ellen Mitchell Elementary School         2233 W Ohio St    
174                 Frederic Chopin Elementary School         2450 W Rice St    
245                  James G Blaine Elementary School        1420 W Grace St    
366  Mary E Courtenay Elementary Language Arts Center     1726 W Berteau Ave    
417                     Oriole Park Elementary School       5424 N Oketo Ave    
488       Stephen Decatur Classical Elementary School  7030 N Sacramento Ave    
494                          Talman Elementary School      5450 S Talman Ave    
555                    William Penn Elementary School       1616 S Avers Ave    

    Grade  SAFETY_SCORE  Instruction Score Teachers Score   Latitude  \
27     ES          99.0               82.0             82  41.697198   
141    ES          99.0               95.0             90  41.892055   
174    ES          86.0               99.0             99  41.896740   
245    ES          99.0               74.0            NDA  41.950808   
366    ES          99.0               80.0            NDA  41.958017   
417    ES          99.0               71.0            NDA  41.978515   
488    ES          99.0               99.0             99  42.009307   
494    ES          99.0               99.0             81  41.794074   
555    ES          78.0               99.0            NDA  41.858370   

     Longitude          borough                     Location  
27  -87.697264  MOUNT GREENWOOD   (41.69719792, -87.6972638)  
141 -87.683179        WEST TOWN  (41.89205482, -87.68317867)  
174 -87.688902        WEST TOWN  (41.89674048, -87.68890154)  
245 -87.664958        LAKE VIEW  (41.95080812, -87.66495825)  
366 -87.671893        LAKE VIEW    (41.9580169, -87.6718933)  
417 -87.812091     NORWOOD PARK  (41.97851501, -87.81209057)  
488 -87.704655       WEST RIDGE  (42.00930696, -87.70465502)  
494 -87.690298        GAGE PARK   (41.79407382, -87.6902983)  
555 -87.721336   NORTH LAWNDALE  (41.85836995, -87.72133609)

In [39]:
df_capstone2_['NAME_OF_SCHOOL'].head()

27         Annie Keller Elementary Gifted Magnet School
141                    Ellen Mitchell Elementary School
174                   Frederic Chopin Elementary School
245                    James G Blaine Elementary School
366    Mary E Courtenay Elementary Language Arts Center
Name: NAME_OF_SCHOOL, dtype: object

In [40]:
chicago_map3 = folium.Map(location=[latitude, longitude], zoom_start=11)

# add markers to map
for NAME_OF_SCHOOL, lat, lng in zip(df_capstone2_['NAME_OF_SCHOOL'], df_capstone2_['Latitude'], df_capstone2_['Longitude']):
    label = '{}'.format(NAME_OF_SCHOOL)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=4,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(chicago_map3)
    
for name, lat2, lng2 in zip (df_capstone4['name'], df_capstone4['lat'], df_capstone4['lng']):
     label1 = '{}'.format(name)                                                          
     label1 = folium.Popup(label1, parse_html=True)                                                           
     folium.CircleMarker(
        [lat2, lng2],
        radius=4,
        popup=label1,
        color='red',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(chicago_map3) 
chicago_map3

# Results  <a name="results"></a>

There are many elementary schools in Chicago about 566 elementary schools in various area communities, A GPS could guide you well or a map.

Our problem was to find a Dance School close to an Elementary School at Chicago and that the elementary school would be a good school by certain criteria such as:

   * SAFETY_SCORE
   * Instruction Score
   * Teachers Score


Enough security of very competent teachers and a very high level of instruction so that the children would be literally well 


After doing our analysis and research through Foursquare, Folium, Clustering and others, there are two Elementary schools with the required criteria that are located near a dance school in Chicago  their names are:
   * Michell Elementary School
   * Chopin Elementary School




# Conclusion <a name="conclusion"></a>

Parents will be able to manage their time when they put their children in an elementary school with a high education score and find a **Dance School** close enough to that **Elementary School** and its **Elementary School** with high education score is good enough to take care of the students' education. Even if there are parents who will not put their children in a ready-made dance school.